logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -226,7 +362,7 @@ void dropTailSegment() throws IOException {
         }
```

Comment:
What would happen if a segment is deleted here? Will this result in an IllegalStateException here?

Reply:
Yes, this has to be fixed. Furthermore, if the listing of segments results in an empty set, the `currentQueueSize`, which is used in the [age policy enforcement loop](https://github.com/elastic/logstash/blob/f073529d70615de92e7d25d7852158494f4173b9/logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java#L207-L209) has to be calculated using the disk physical data with method `computeQueueSize()`

Reply:
Fixed by 837884dccfc1371167d914a94589a2f821601ea2